### DISCUSSION TITLE
Change LWT text and the location.

### DISCUSSION THREAD

**Discussion by EDSTOBI** [arendst/Tasmota#15373] [Q&A] [answered]
Hello
I already have a lot of devices running into my MQTT and with Node Red a monitoring on LWT.
With my new 4 shelly 1PM I now have the problem that I want to change the LWT text and the location.
Lik that:
[ stateTopic="LWT Topic/%hostname%",on="Bin da",off="HELP ME!" ] 

Can I do this with Tasmota or is this only possible with espeasy?

Furthermore, I wonder how I can adjust Info1-3. That I have as a new Tasmota user not yet herrausgefunden

THX

**Comment by barbudor** [comment, marked as answer]
As of today, what is published in LWT and INFO1-3 are fixed and not configurable.
You can always publish anything you want using rule and Publish (non retained) or Publish2 (retained)

**Reply by sfromis** [reply]
You can actually influence the content of the LWT message using `SetOption10` or at at build time.
```
#define MQTT_LWT_MESSAGE       false             // [SetOption10] Switch between MQTT LWT OFFLINE or empty message
#define MQTT_LWT_OFFLINE       "Offline"         // MQTT LWT offline topic message
#define MQTT_LWT_ONLINE        "Online"          // MQTT LWT online topic message
```

**Reply by EDSTOBI** [reply]
Thank you for the answer.
Do I understand correctly that I can change "offline" and "online" with define? This does not work for me, I can only activate or deactivate the LWT, but not adjust the text.  if it works, how do I enter the setoption10 in the console?

**Reply by barbudor** [reply]
#define are compile option. So you need to compile your own firmware to change the value
Which means the value are hardcoded in the firmware and if different devices need to publish different message (like including their hostname and location) you need a different firmware for each device.

SetOption10 only controls Tasmota behavior if you use the command "Topic" to change the current topic value to another one : should it say goodbye properly or not on the current topic before restarting on the new one.

Still not really understanding what you want to achieve with changing the value of the LWT message.
I'm pretty confident that you would achieve similar by pushing your own message with the Publish/Publish2 command in a boot-rule.

**Reply by EDSTOBI** [reply]
Hello 
Pure and simple follyness.
I have almost 100 ESPEASY running.
These have formed a solid structure in the mqtt over the last few years.
I have a monitoring in nodered which is partly very complex. I had the 4 shelly tasmota like to pack into this mqtt structure.
Espeasy is not yet so far advanced with current measurement in shelly1pm, so I have switched to tesmota.
THX, I will adjust my monitoring or wait for espeasy
